ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -175,7 +175,12 @@ Tensor to_layout_impl(
                 return ttnn::tilize(tensor, output_memory_config, dtype, use_multicore_tilize);
             } else {
                 tensor = ttnn::tilize_with_val_padding(
-                    tensor, padded_output_shape, 0, output_memory_config, dtype, use_multicore_tilize);
+                    tensor, padded_output_shape,
+                    (tensor.get_dtype() == ttnn::DataType::BFLOAT16 or tensor.get_dtype() == ttnn::DataType::FLOAT32) ? (float) 0.0 : (int) 0,
```

Comment:
This is a bug, this will always pass the float value to the variant. You have another instance of this below. This is because the ternary operator has to return the same type from both branches, and here an int will always be promoted to a float.